build/Targets/VSL.Versions.targets

```diff
@@ -40,32 +48,32 @@
 
   <PropertyGroup>
     <!-- Split the build parts out from the BuildNumber which is given to us by MicroBuild in the format of yyyymmdd.nn
-         where BuildNumberPart1 is yyyymmdd (such as 20160615) and BuildNumberPart2 is nn (which represents the nth build 
-         started that day). So the first build of the day, 20160615.1, will produce something similar to BuildNumberPart1: 20160615,
-         BuildNumberPart2: 01; and the 12th build of the day, 20160615.12, will produce BuildNumberPart1: 20160615, BuildNumberPart2: 12 -->
-    <!-- Additionally, in order to match the MicroBuild v1 behavior, we will only take the last five digits of the BuildNumber, so
-         in the case of 20160615, we will set BuildNumberPart1 to 60615. This will begin failing in the 2017 as the BuildVersion only allows
+         where BuildNumberFiveDigitDateStamp is yyyymmdd (such as 20160615) and BuildNumberBuildOfTheDay is nn (which represents the nth build 
+         started that day). So the first build of the day, 20160615.1, will produce something similar to BuildNumberFiveDigitDateStamp: 60615,
+         BuildNumberBuildOfTheDayPadded: 01; and the 12th build of the day, 20160615.12, will produce BuildNumberFiveDigitDateStamp: 20160615, BuildNumberBuildOfTheDay: 12 -->
+    <!-- Additionally, in order ensure the value fits in the 16-bit PE header fields, we will only take the last five digits of the BuildNumber, so
+         in the case of 20160615, we will set BuildNumberFiveDigitDateStamp to 60615. This will begin failing in the 2017 as the BuildVersion only allows
          each part to be in the range of 0 through 65535. Issue #12038 tracks the fix that needs to happen. -->
-    <BuildNumberPart1>$(BuildNumber.Split('.')[0].Substring(3))</BuildNumberPart1>
-    <BuildNumberPart2>$(BuildNumber.Split('.')[1].PadLeft(2,'0'))</BuildNumberPart2>
+    <BuildNumberFiveDigitDateStamp>$(BuildNumber.Split('.')[0].Substring(3).Trim())</BuildNumberFiveDigitDateStamp>
+    <BuildNumberBuildOfTheDayPadded>$(BuildNumber.Split('.')[1].PadLeft(2,'0'))</BuildNumberBuildOfTheDay>
   </PropertyGroup>
   
   <Choose>
     <When Condition="'$(BuildVersion)' != ''">
       <!-- The user specified a build version number. In that case, we'll use their version number
-          for the file version, and force the assembly version to $(RoslynSemanticVersion).0.  That way
+          for the file version, and force the assembly version to $(RoslynAssemblyVersionBase).0.  That way
           day-to-day upgrades don't break assembly references to other installed apps. -->
       <PropertyGroup>
-        <AssemblyVersion>$(RoslynSemanticVersion).0</AssemblyVersion>
-        <VsixVersion>$(RoslynSemanticVersion).$(BuildNumberPart1.Trim())$(BuildNumberPart2.Trim())</VsixVersion>
+        <AssemblyVersion>$(RoslynAssemblyVersionBase).0</AssemblyVersion>
+        <VsixVersion>$(RoslynFileVersionBase).$(BuildNumberFiveDigitDateStamp)$(BuildNumberBuildOfTheDayPadded)</VsixVersion>
```

Comment:
Not introduced by your change, but I this means that builds that specify `BuildVersion` but not `BuildNumber` have very strange vsix versions.  Since we're going to embed `BuildVersion` into the file version attribute, should we just use it here for `VsixVersion` too?